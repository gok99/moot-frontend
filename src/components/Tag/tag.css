.tag {
  background: #7e5ab8;
  border-radius: 12px;
  padding: 0;
  padding-left: 10px;
  margin: 0px 2px;
}

.btn-tag,
.btn-tag:visited,
.btn-tag:default,
.btn-tag:focus {
  background: #5f3d96;
  color: #ffffff;
  border: none;
  height: 24px;
  width: 32px;
  margin-left: 8px;
  border-radius: 0px 12px 12px 0px;
  text-align: center;
  padding: 0;
  padding-bottom: 28px;
}

.btn-tag:disabled {
  background: #5f3d96;
  color: #ffffff;
  border: none;
  opacity: 0.4 !important;
}

.btn-tag:hover,
.btn-tag:focus:hover {
  background: #4f2d85;
  color: #ffffff;
  border: none;
}

.btn-tag:focus:active,
.btn-tag:active {
  background-color: #31165c;
  color: #ffffff;
  border: none;
}
