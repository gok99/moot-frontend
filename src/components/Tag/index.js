import React from 'react';
import { Badge, Button } from 'react-bootstrap';

import './tag.css';

const Tag = ({ tagName, owned, onTagPress, postCreationCheck }) => {
  var ownedState = owned;
  const onClick = (event) => {
    // This function should be one that adds / removes tags, takes in a boolean: if true, remove; if false, add.
    (onTagPress(ownedState, tagName))(event); 
    ownedState = !ownedState;
  };
  const onClickPostCreation = (event) => {
    (onTagPress(tagName))(event);
  }

  return (
    <Badge pill className="tag mb-1" bg="primary">
      {tagName}
      <Button className="btn-tag" onClick={postCreationCheck ? onClickPostCreation : onClick}>
        { ownedState ? "-" : "+" }
      </Button>
    </Badge>
  );
};

export default Tag;