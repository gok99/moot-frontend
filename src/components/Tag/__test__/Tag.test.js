import React from 'react';
import ReactDOM from 'react-dom';
import Tag from '../index';

import { cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<Tag tagName={""} owned={true} onTagPress={(tag) => (event) => { event.preventDefault(); }} postCreationCheck={true} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<Tag tagName={""} owned={false} onTagPress={(tag) => (event) => { event.preventDefault(); }} postCreationCheck={true} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<Tag tagName={""} owned={true} onTagPress={(tag) => (event) => { event.preventDefault(); }} postCreationCheck={false} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<Tag tagName={""} owned={false} onTagPress={(tag) => (event) => { event.preventDefault(); }} postCreationCheck={false} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<Tag tagName={""} owned={true} onTagPress={(tag) => (event) => { event.preventDefault(); }} postCreationCheck={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot", () => {
  const tree = renderer.create(<Tag tagName={""} owned={false} onTagPress={(tag) => (event) => { event.preventDefault(); }} postCreationCheck={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot", () => {
  const tree = renderer.create(<Tag tagName={""} owned={true} onTagPress={(tag) => (event) => { event.preventDefault(); }} postCreationCheck={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot", () => {
  const tree = renderer.create(<Tag tagName={""} owned={false} onTagPress={(tag) => (event) => { event.preventDefault(); }} postCreationCheck={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});