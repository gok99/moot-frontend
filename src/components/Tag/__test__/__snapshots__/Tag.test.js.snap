// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<span
  bg="primary"
  className="tag mb-1 badge badge-pill"
>
  
  <button
    className="btn-tag btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    -
  </button>
</span>
`;

exports[`matches snapshot 2`] = `
<span
  bg="primary"
  className="tag mb-1 badge badge-pill"
>
  
  <button
    className="btn-tag btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    +
  </button>
</span>
`;

exports[`matches snapshot 3`] = `
<span
  bg="primary"
  className="tag mb-1 badge badge-pill"
>
  
  <button
    className="btn-tag btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    -
  </button>
</span>
`;

exports[`matches snapshot 4`] = `
<span
  bg="primary"
  className="tag mb-1 badge badge-pill"
>
  
  <button
    className="btn-tag btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    +
  </button>
</span>
`;
